packages/react/menu/src/Menu.tsx

```diff
@@ -64,11 +65,13 @@ Menu.displayName = MENU_NAME;
 
 const CONTENT_NAME = 'MenuContent';
 
-type MenuItemRef = React.RefObject<React.ElementRef<typeof MenuItem>>;
-type MenuContentContextValue = {
-  itemMap: Map<MenuItemRef, { ref: MenuItemRef; disabled: boolean }>;
-  onItemLeave(): void;
-};
+type ItemData = { disabled: boolean };
+const [CollectionSlot, CollectionItemSlot, useCollection] = createCollection<
+  HTMLSpanElement,
```

Comment:
Is there a way to get this dynamically from the type of element actually defined in the menu item rather than duplicating here with the risk of getting out of sync?

Reply:
Do you mean `React.ElementRef<typeof MenuItem>` ?

Reply:
of course! why do I keep forgetting about this?? 😀